packages/components/src/components/form/SegmentSlider/SegmentSlider.tsx

```diff
@@ -0,0 +1,242 @@
+    margin: ${spacingsPx.sm} 0 ${spacingsPx.xs};
+    padding: 10px 0;
+    width: 100%;
+    vertical-align: top; /* prevent extra bottom space in Firefox */
+
+    position: relative;
+    z-index: 10;
+
+    width: 100%;
+
+    background: none;
+    appearance: none;
+    cursor: ${({ disabled }) => !disabled && 'pointer'};
+
+    &::-webkit-slider-runnable-track {
+        appearance: none;
+        background: transparent;
+    }
+
+    &::-webkit-slider-thumb {
+        ${thumb};
+    }
+
+    &::-moz-range-track {
+        appearance: none;
+        background: transparent;
+    }
+
+    &::-moz-range-thumb {
+        ${thumb};
+        transform: translateY(-6px); /* Firefox hack because of position absolute */
+    }
+
+    &:disabled {
+        pointer-events: auto;
+        cursor: not-allowed;
+    }
+`;
+
+const Segments = styled.div`
+    position: absolute;
+    top: 0;
+
+    margin: ${spacingsPx.sm} 0 ${spacingsPx.xs};
+    padding: 10px 0;
+    width: 100%;
+`;
+
+const Segment = styled.div<{ $start: number; $end: number }>`
+    --segment-padding: 2px;
+
+    position: absolute;
+    left: ${({ $start }) => $start}%;
+    width: calc(${({ $start, $end }) => $end - $start}% - var(--segment-padding) * 2);
+`;
+
+const SegmentLine = styled.div<{ $filled: number; disabled?: boolean }>`
+    height: ${spacingsPx.xxs};
+    background: ${({ $filled, theme, disabled }) =>
+        `linear-gradient(90deg, ${disabled ? theme.backgroundNeutralDisabled : theme.backgroundPrimaryDefault} ${$filled}%, ${disabled ? theme.backgroundNeutralDisabled : theme.backgroundNeutralSubdued} ${$filled}%)`};
+
+    border-radius: ${borders.radii.full};
+`;
+
+const SegmentLabel = styled.div`
+    margin-top: ${spacingsPx.sm};
+    padding-top: ${spacingsPx.xxxs};
+    text-align: left;
+    color: ${({ theme }) => theme.textSubdued};
+    ${typography.label}
+
+    span {
+        cursor: pointer;
+    }
+`;
+
+type Segment = {
+    max: number;
+    name: string;
+};
+
+const normalizeValue = (value: number, min: number, max: number) =>
+    ((value - min) / (max - min)) * 100;
+
+const getPreviousSegment = (segments: Segment[], index: number): Segment => {
+    if (index <= 0) return DEFAULT_SEGMENT;
+    return segments.at(index - 1) ?? DEFAULT_SEGMENT;
+};
+
+const getProgress = (
+    value: number,
+    segment: Segment,
+    previousSegment: Segment,
+    sliderMin: number,
+    sliderMax: number,
+): number => {
+    const isActive = value <= segment.max && value > previousSegment.max;
+    const normalizedPreviousMax = normalizeValue(previousSegment.max, sliderMin, sliderMax);
+    const normalizedSegmentMax = normalizeValue(segment.max, sliderMin, sliderMax);
+    return isActive
+        ? ((normalizeValue(value, sliderMin, sliderMax) - normalizedPreviousMax) /
+              (normalizedSegmentMax - normalizedPreviousMax)) *
+              100
+        : value > segment.max
+          ? 100
+          : 0;
```

Comment:
Eslint error
`Do not nest ternary expressions.eslint[no-nested-ternary](https://eslint.org/docs/latest/rules/no-nested-ternary)`